src/app/linux.rs

```diff
@@ -1,20 +1,180 @@
 // vim: tw=80
-use std::error::Error;
+
+#![warn(clippy::all, clippy::pedantic)]
+
+use std::{error::Error, fs::File, io, io::BufRead, path::PathBuf};
+
+use glob::glob;
 
 use super::Snapshot;
 
-pub(super) struct SnapshotIter {}
+// Similar to sysctl::CtlValue, but only as many types as necessary.
+#[derive(Debug)]
+enum ObjsetValue {
+    String(String),
+    U64(u64),
+}
+
+fn parse_objset_row(row: &str) -> Option<(String, ObjsetValue)> {
+    let mut fields = row.split_ascii_whitespace();
+
+    match (fields.next(), fields.next(), fields.next()) {
+        (Some(name), Some(_), Some(value)) => {
+            let field_name = (*name).to_string();
+            if field_name == "dataset_name" {
+                Some((field_name, ObjsetValue::String((*value).to_string())))
+            } else {
+                match value.parse::<u64>().ok() {
+                    Some(n) => Some((field_name, ObjsetValue::U64(n))),
+                    None => Some((
+                        field_name,
+                        ObjsetValue::String((*value).to_string()),
+                    )),
+                }
+            }
+        }
+        _ => None,
+    }
+}
+
+fn parse_objset<R: BufRead>(reader: R) -> io::Result<Snapshot> {
+    // The first line contains raw numeric data that we don't collect.
+    // The second line contains column headers. Both these lines are skipped.
+    let lines = reader.lines().skip(2);
+
+    let mut snap = Snapshot {
+        name:      String::new(), // temporary value will be overwritten
+        nread:     0,
+        nwritten:  0,
+        nunlinks:  0,
+        nunlinked: 0,
+        reads:     0,
+        writes:    0,
+    };
```

Comment:
Tested and working on debian bullseye + nixos

One small suggestion:
```suggestion
    let mut snap = Snapshot::default();
```
This can be replaced if `#[derive(Default)]` is added to `Snapshot`, they are functionally the same.

Reply:
Apologies. This was implemented, but never resolved the thread.